### PR TITLE
feat(dashboard-layout): dashboard widgetFrame width assigning algorithm

### PR DESCRIPTION
**Pull request by minseolee**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [ ] `Error / Warning / Lint / Type`

### Description

WIP
- [x] Debug some bugs
- [x] Write test codes
- [x] Add more specific annotations
- [x] Change Korean annotations to English
- [x] Support mobile (screen <= 720px)

스크린샷 2022-11-22 오후 8 53 03

스크린샷 2022-11-22 오후 8 53 10

### CODE REVIEW

File: src/services/dashboards/dashboard-detail/lib/helper.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    for (let i = 0; i < eachLineCardList.length; i += 1) {
+        const oneLineSequence: Array<number> = [];
+        for (let j = 0; j < eachLineCardList[i].length; j += 1) {
+            oneLineSequence.push(0);
+        }
+        eachLineSequenceList.push(oneLineSequence);
+    }
+
+    // i -> 각 row 순회
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/lib/helper.ts:60]: 😲

File: src/services/dashboards/dashboard-detail/lib/helper.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+/** NAMING
+ * widgetFrameSize: string -> 'SM'|'MD'|'LG'|'XL'
+ * widgetFrameWidth: number -> 320, 400... 880, 960
+ * widgetFrameWidthRange: Array<number> -> [320, 400, 480]|[480, 560, 640]|[800, 880, 960]|[800, 880, 960]
+ * rowWidgetFrameWidthRange: Array<Array<number>> -> [[MD_RANGE], [MD_RANGE], [SM_RANGE], [LG_RANGE]]
+ * allWidgetFrameWidthRange: Array<Array<Array<number>>> -> [[[MD_RANGE], [MD_RANGE]], [[SM_RANGE], [[LG_RANGE]]]
+ * rowWidgetFrameWidthList: Array<number> -> [320, 480]
+ * allWidgetFrameWidthList: Array<Array<number>> -> [[320], [320, 480], [320, 480]]
+ * */
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/lib/helper.ts:15]: Thanks for the comment.

File: src/services/dashboards/dashboard-detail/lib/config.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// [480, 560, 640, 640, 640]
+// [640, 720, 800, 800, 800]
+// [800, 880, 960, 960, 960
+
+export const WIDGET_FRAME_WIDTH_RANGE_LIST = [
+    [320, 400, 480], // 0: SM
+    [480, 560, 640], // 1: MD
+    [640, 720, 800], // 2: LG
+    [800, 880, 960], // 3: XL
+];
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/lib/config.ts:16]: It doesn't seem to have to be an array, how about changing it like this?
Then it seems that you don't have to search by number index in `widgetFrameSizeRangeExtractor()`.

```suggestion
export const WIDGET_FRAME_WIDTH_RANGE = {
    SM: [320, 400, 480],
    MD: [480, 560, 640],
    LG: [640, 720, 800],
    XL: [800, 880, 960],
};
```

---
꼭 array일 필요는 없어 보이는데 이렇게 바꾸는 건 어떨까요?
그럼 widgetFrameSizeRangeExtractor에서  number index로 찾지 않아도 될 것 같습니다.

**Review by WANZARGEN — APPROVED** [review]
What a hard work..🥲 
We already did code review in offline, so I just checked conventions.

File: src/services/dashboards/dashboard-detail/lib/config.ts (hunk 3 above)
Comment (minseolee) [reply src/services/dashboards/dashboard-detail/lib/config.ts:16]: Great! I'll apply it. Thanks a lot

File: .eslintrc.cjs [hunk 4]
```diff
@@ -35,8 +35,9 @@ module.exports = {
         'no-unused-vars': 1,
         'no-mixed-operators': 0,
         'no-promise-executor-return': 1,
+        'no-multiple-empty-lines': 0,
         'prefer-regex-literals': 1,
-        'prefer-const': 1,
+        'prefer-const': 0,
```
[the remark this answers is not in the document]
Comment (minseolee) [reply .eslintrc.cjs:40]: While writing this code, I've used `let` a lot.
But eslint often changed the `let` to `const`. 
I'll check it one more time

**Pull request merged**